plenum/server/node.py

```diff
@@ -2354,15 +2354,42 @@ def lost_master_primary(self):
         self._schedule_view_change()
 
     def select_primaries(self, nodeReg: Dict[str, HA]=None):
+        primaries = set()
+        primary_rank = None
+        '''
+        Build a set of names of primaries, it is needed to avoid
+        duplicates of primary nodes for different replicas.
+        '''
+        for instance_id, replica in enumerate(self.replicas):
```

Comment:
Do we need to keep track of 'old' primaries just to avoid selecting the same primaries as was before?
Please add a test for this!